clang/include/clang/Serialization/ASTReader.h

```diff
@@ -654,6 +654,10 @@ class ASTReader
   /// Map from the TU to its lexical contents from each module file.
   std::vector<std::pair<ModuleFile*, LexicalContents>> TULexicalDecls;
 
+  unsigned HeaderFileInfoIdx = 0;
```

Comment:
nit: Unloaded header file info idx.

And also I feel the name is odd. It is index for module files, not header file infos.

Reply:
And also, this seems unsafe since ModuleManager can remove modules technically. Maybe it is better to insert a bool in ModuleFile to mark if its header info is loaded.